### ISSUE TITLE
sferes.create_variants also compiles source file without any definitions

### ISSUE THREAD

**Issue by JoostHuizinga** [sferes2/sferes2#48]
This is a difference between old sferes behavior and new sferes behavior. When creating the following wscript file in an experiment:

```
#! /usr/bin/env python
import sferes

def options(opt): pass

def build(bld):
   uselib_ = 'BOOST EIGEN SDL'
   sferes.create_variants(bld,
                             source = 'maze.cpp',
                             use = 'sferes2 fastsim',
                             target = '',
                             uselib = 'BOOST EIGEN SDL',
                             variants = ['GEN1 POP4'])
```

My old sferes version compiles `maze_gen21_pop4.cpp` with `GEN1` and `POP4` defined. However, the current sferes version also tries to compile `maze.cpp`, without any defines. This is a problem for my current experimental setup, since many of my 'variant' arguments are mandatory, and `maze.cpp` does not compile as a result (and `maze_gen21_pop4.cpp` does not finish compilation either, because `maze.cpp` fails).

As such, I am curious what the intended behavior is of `sferes.create_variants`, and what the right way is to deal with this issue. For now, I have removed the statement that causes create_variants to build `maze.cpp` in my local copy.

**Comment by costashatz** [comment]
You are right about that. We need to change it. I will.

**Comment by costashatz** [comment]
Solved in #53 ...
